### PR TITLE
Enhance dump command to show models

### PR DESCRIPTION
**Pull request by mavam**
###  :notebook_with_decorative_cover: Description

This PR enhances the `dump` command to show models in addition to concepts.

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [ ] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.
- [ ] Address FIXMEs (empty-string `description` when dumping)

### :dart: Review Instructions

@dominiklohmann:
  - [x] All command-related C++ file-by-file

@tobim:
  - [x] Taxonomy map storage change
  - [x] Integration test suite changes

### CODE REVIEW

File: libvast/src/system/application.cpp [hunk 1]
```diff
@@ -134,6 +134,9 @@ auto make_dump_command() {
   dump->add_subcommand("concepts", "print all registered concept definitions",
                        documentation::vast_dump_concepts,
                        opts("?vast.dump.concepts"));
+  dump->add_subcommand("models", "print all registered model definitions",
+                       documentation::vast_dump_concepts,
+                       opts("?vast.dump.concepts"));
```
Comment (dominiklohmann) [inline libvast/src/system/application.cpp:139]:
```suggestion
  dump->add_subcommand("models", "print all registered model definitions",
                       documentation::vast_dump_models,
                       opts("?vast.dump.models"));
```

And then the command docs are also missing.

**Review by dominiklohmann — CHANGES_REQUESTED** [review]

**Comment by mavam** [comment]
As the guardian of the integration test suite, @tobim may want to have a look at https://github.com/tenzir/vast/pull/1233/commits/89152f1485f063714f9cf3089e3b2d09498fdf4e and https://github.com/tenzir/vast/pull/1233/commits/9ecbfc889402ce0b9bd2af31d3e86cfa38ae081e.

**Review by dominiklohmann — APPROVED** [review]
Code-wise this is fine.

Can you change the documentation of `vast dump` itself so it reflects that it dumps both models and concepts?

Please wait for @tobim to review the integration test suite changes. I did not look at them.

**Comment by mavam** [comment]
I'm handing this PR over to @tobim, who is going to investigate the non-determinism during concept dumping.

**Comment by mavam** [comment]
Taking the PR back after internal sync.

File: integration/integration.py [hunk 2]
```diff
@@ -192,15 +192,9 @@ def empty(iterable):
...
+def is_non_deterministic(command):
     positionals = list(filter(lambda x: x[0] != "-", command))
-    if positionals[0] == "export":
-        if positionals[1] == "json":
-            # Pythons JSON parser does not handle ldjson, so we fall back to ascii
-            return "ascii"
-    if positionals[0] == "status":
-        return "json"
-    return "ascii"
+    return bool(re.match(r"exp(ort|lore)|get|pivot", positionals[0]))
```
Comment (tobim) [inline integration/integration.py:197]:
```suggestion
    return positionals[0] in {"export", "explore", "get", "pivot"}
```

File: integration/integration.py [hunk 3]
```diff
@@ -11,6 +11,7 @@
 import json
 import logging
 import os
+import re
```
Comment (tobim) [inline integration/integration.py:14]:
```suggestion
```

**Review by tobim — CHANGES_REQUESTED** [review]
Just a suggestions to simplify the python code.

**Comment by mavam** [comment]
> You only take a const std::regex& here, so you should get away with forward-declaring it.

Is there a header for this, like we have for `<iosfwd>`?

> `<regex>` is one of the most expensive headers to include, and I'm not sure why we need it here. 

It makes for a convenient way to control the size of the return value.

>The default value is also crazy expensive. std::regex should never be created on the fly.

We're doing I/O here. And it's happening pretty much once at startup. The performance concerns don't appear to be in relation to the functionality.

**Review by dominiklohmann — APPROVED** [review]
This works fine.

Not printing empty descriptions / nested concepts / fields is a strict follow up and should not block the imminent release.

The `<regex>` stuff I'll clean up myself after the release, I created a story for that.

**Pull request merged**